src/ILLink.RoslynAnalyzer/DataFlow/LocalDataFlowVisitor.cs

```diff
@@ -199,7 +198,14 @@ public override TValue VisitSimpleAssignment (ISimpleAssignmentOperation operati
 						// For now, just don't warn. https://github.com/dotnet/linker/issues/2731
 						break;
 					}
-					HandleMethodCall (setMethod, instanceValue, ImmutableArray.Create (value), operation);
+
+					// Property may be an indexer, in which case there will be one or more index arguments followed by a value argument
```

Comment:
Definitely a good fix, but I would prefer to do this in a separate PR - along with some tests for this.

Reply:
Moved those changes to https://github.com/dotnet/linker/pull/3081